src/reducers/versions.tsx

```diff
@@ -686,13 +686,37 @@ const reducer: Reducer<VersionsState, ActionType<typeof actions>> = (
     case getType(actions.updateSelectedPath): {
       const { selectedPath, versionId } = action.payload;
 
+      const version = state.versionInfo[versionId];
+      const { expandedPaths } = version;
+
+      // Make sure the root folder is expanded.
+      const parents = [];
+      const rootPath = getRootPath(version);
+      if (!expandedPaths.includes(rootPath)) {
+        parents.push(rootPath);
+      }
+
+      const folders = selectedPath.split('/');
+
+      while (folders.length > 1) {
+        folders.pop();
+        const newPath = folders.join('/');
+        if (!expandedPaths.includes(newPath)) {
+          parents.push(newPath);
+        }
+      }
+
       return {
         ...state,
         versionInfo: {
           ...state.versionInfo,
           [versionId]: {
             ...state.versionInfo[versionId],
             selectedPath,
+            expandedPaths: [
+              ...state.versionInfo[versionId].expandedPaths,
+              ...parents,
```

Comment:
How about this?

```js
...parents.filter((newPath) => !expandedPaths.includes(newPath))
```

With that, I think you won't have to duplicate the checks up above.

Reply:
Nice! Yes, that's better. 👍